code/nebula/neb.cpp

```diff
@@ -191,62 +201,109 @@ void neb2_init()
 
 		// poofs
 		while (required_string_one_of(3, "#end", "+Poof:", "$Name:")) {
+			bool create_if_not_found = true;
+			poof_info pooft;
+			poof_info* poofp;
+			bool poof_new = true;
 
-			if (Poof_info.size() < MAX_NEB2_POOFS) {
-				poof_info new_poof;
+			if (optional_string("+Poof:")) { // retail style
+				stuff_string(name, F_NAME, MAX_FILENAME_LEN);
+				strcpy_s(pooft.bitmap_filename, name);
 
-				if (optional_string("+Poof:")) { // retail style
-					stuff_string(name, F_NAME, MAX_FILENAME_LEN);
-					strcpy_s(new_poof.bitmap_filename, name);
+				strcpy_s(pooft.name, name);
 
-					strcpy_s(new_poof.name, name);
+				generic_anim_init(&pooft.bitmap, name);
 
-					generic_anim_init(&new_poof.bitmap, name);
+				if (Poof_info.size() < MAX_NEB2_POOFS) {
+					Poof_info.push_back(pooft);
+				} else {
+					Warning(LOCATION, "More than 32 poofs are defined! Skipping poof %s\n", pooft.name);
+				}
+			} else if (optional_string("$Name:")) { // new style
+				stuff_string(pooft.name, F_NAME, NAME_LENGTH);
+
+				if (optional_string("+nocreate")) {
+					if (!Parsing_modular_table) {
+						Warning(LOCATION, "+nocreate flag used for nebula poof in non-modular table\n");
+					}
+					create_if_not_found = false;
+				}
 
-					Poof_info.push_back(new_poof);
-				} else if (optional_string("$Name:")) { // new style
-					stuff_string(new_poof.name, F_NAME, NAME_LENGTH);
+				// Does this poof exist already?
+				// If so, load this new info into it
+				poofp = get_nebula_poof_pointer(pooft.name);
+				if (poofp != NULL) {
+					if (!Parsing_modular_table) {
+						error_display(1,
+							"Error:  Nebula Poof %s already exists.  All nebula poof names must be unique.",
+							pooft.name);
+					}
+					poof_new = false;
+				} else {
+					// Don't create poof if it has +nocreate and is in a modular table.
+					if (!create_if_not_found && Parsing_modular_table) {
+						if (!skip_to_start_of_string_either("$Name:", "#end")) {
+							error_display(1, "Missing [#end] or [$Name] after nebula poof %s", pooft.name);
+						}
+						continue;
+					}
+					// Check if we're at max poofs. If so, then log and continue.
+					if (Poof_info.size() < MAX_NEB2_POOFS) {
+						Poof_info.push_back(pooft);
+						poofp = &Poof_info[Poof_info.size() - 1];
+					} else {
+						Warning(LOCATION, "More than 32 poofs are defined! Skipping poof %s\n", pooft.name);
+						if (!skip_to_start_of_string_either("$Name:", "#end")) {
+							error_display(1, "Missing [#end] or [$Name] after nebula poof %s", pooft.name);
+						}
+						continue;
+					}
+				}
 
+				if (poof_new) {
 					required_string("$Bitmap:");
 					stuff_string(name, F_NAME, MAX_FILENAME_LEN);
+				} else {
+					optional_string("$Bitmap:");
+					stuff_string(name, F_NAME, MAX_FILENAME_LEN);
```

Comment:
Are you wanting to prevent someone adding `$Bitmap: ` with no string defined afterwards here? (Essentially trying to clear the bitmap in a tbm?)

Reply:
The opposite.
Right now, you could not specify $Bitmap (if its a nocreate entry), but it would still try to parse the filename after the nonexistant token

Reply:
Oh. Right. I see what you're saying now. Yeah. Will do.